Added message scrolling, ACP and global time variables.

judge2005 opened the pull request:
Hi,

I thought you might be interested in several things I added to your base code:

1. ACP every 10 minutes (function is called oneArmedBandit()).
2. Display the date very minute at 50s - this uses a class called Transition. I implemented several transitions - scrolling and scrambling. This also means I had to add an extra mechanism to blank digits.
3. Use global variables for hours, minutes, seconds and milliseconds. This is so that all the functions called from loop() see the same values. For example, this means that the flashing separators stay in sync with the actual seconds display.

I hope you find these changes interesting.

Thanks - Paul Andrews

judge2005 committed:
Added message scrolling, ACP and global time variables.

judge2005 committed:
Implemented blanking and fixed error in tune cancellation.

judge2005 committed:
Implement leading zero blanking for time.

judge2005 committed:
Add date format selection

afch commented:
I read all of your messages, but I'm on vacation and I have no way to check it out. But anyway, I really appreciate what you are doing!

judge2005 committed:
The blue LED wasn't cycling because PWM on pins 3 and 11 is driven by
timer2, which is what the tone library uses. Re-implemented color
cycling using HSV model and added code to re-enable PWM on 3 and 11 when
no tune is being played.

Also fixed problem with blanking if end hour < start hour!

judge2005 commented:
No problem. Love the clock and you are welcome to the code if you want to use it. Compiled code size is quite large now (though it still fits!). It could be reduced by reducing the number of transition types.

If you want to discuss any of the changes, feel free to get in touch.

afch commented:
I just looked at all your commits. Your code are awsome, you are really pro!
We made a new shield rev 1.2 that support:
Arduino Mega (and UNO too),
Temperature sensor (work - tested but not implemented yet in firmware)
IR sensor (work - tested but but not implemented yet in firmware)
Light Sensor (not tested)
GPS (only for Arduino Mega, work - tested but but not implemented yet in firmware)
We want to present for You this new shield for 1$(+shipping cost):
just make "best offer" on 1$ for this listing: http://www.ebay.com/itm/162198121548

afch commented:
I have a question: why You decide to move on HSV color model?

judge2005 committed:
Added WiFi code.

judge2005 committed:
Fixed a few issues. Particularly with the HSV calcuation - it needed to
use unsigned ints!

judge2005 committed:
Default to using RTC on startup in case WiFi is down.

judge2005 committed:
Allow RTC to be updated via buttons even in wifi mode

judge2005 committed:
Add extra commands to documentation.

judge2005 committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	NixieClockShieldNCS314.ino

judge2005 committed:
Merge all changes into master

This includes support for WiFi but #ifdef'd out. Same for 1.2 hardware.

judge2005 committed:
Add dimming and cross fading

judge2005 committed:
Tweaks to dimming code and fix music silencing during test.

judge2005 committed:
Fix latching and optimize doTest()

judge2005 committed:
Handle 1.1 and 1.2 hardware

judge2005 committed:
Change some variables from long to unsigned long

judge2005 committed:
Rebase from upstream

judge2005 committed:
Merge remote-tracking branch 'origin/master'
